### PR TITLE
Violation Exception Option

### PR DESCRIPTION
**Pull request by natestuder**
Optionally cause exceptions on policy violations instead of just crashing QEMU.

### CODE REVIEW

**Comment by natestuder** [comment]
Reworked to use the new IR validator integration.

**Comment by natestuder** [comment]
Can we get this resurrected version merged?

**Comment by baltazarortiz** [comment]
I don't know the QEMU stuff well enough to comment on details, but once I get there in not too far in the future I'll make the reference model do the same or very similar to whatever you do here!

File: target/riscv/cpu_bits.h [hunk 1]
```diff
@@ -495,6 +495,7 @@
 #define RISCV_EXCP_INST_PAGE_FAULT         0xc /* since: priv-1.10.0 */
 #define RISCV_EXCP_LOAD_PAGE_FAULT         0xd /* since: priv-1.10.0 */
 #define RISCV_EXCP_STORE_PAGE_FAULT        0xf /* since: priv-1.10.0 */
+#define RISCV_POLICY_VIOLATION_FAULT       0x11 /* RESERVED */
```
Comment (arunthomas) [inline target/riscv/cpu_bits.h:498]: Might be better to use 24–31 or 48–63, since those exception codes are available for custom use.

**Pull request merged**